lib/ts/recipe/thirdpartyemailpassword/recipe.ts

```diff
@@ -0,0 +1,441 @@
+                            });
+                        } else {
+                            return await this.config.signUpFeature.handlePostSignUp(user, {
+                                loginType: "thirdparty",
+                                thirdPartyAuthCodeResponse,
+                            });
+                        }
+                    },
+                },
+                signOutFeature: {
+                    disableDefaultImplementation: this.config.signOutFeature.disableDefaultImplementation,
+                },
+                emailVerificationFeature: {
+                    disableDefaultImplementation: true,
+                },
+            })(appInfo) as ThirdPartyRecipe;
+        }
+        this.emailVerificationRecipe = new EmailVerificationRecipe(
+            recipeId,
+            appInfo,
+            this.config.emailVerificationFeature
+        );
+    }
+
+    static init(config: TypeInput): RecipeListFunction {
+        return (appInfo) => {
+            if (Recipe.instance === undefined) {
+                Recipe.instance = new Recipe(Recipe.RECIPE_ID, appInfo, config);
+                return Recipe.instance;
+            } else {
+                throw new STError(
+                    {
+                        type: STError.GENERAL_ERROR,
+                        payload: new Error(
+                            "ThirdParty recipe has already been initialised. Please check your code for bugs."
+                        ),
+                    },
+                    Recipe.RECIPE_ID
+                );
+            }
+        };
+    }
+
+    static reset() {
+        if (process.env.TEST_MODE !== "testing") {
+            throw new STError(
+                {
+                    type: STError.GENERAL_ERROR,
+                    payload: new Error("calling testing function in non testing env"),
+                },
+                Recipe.RECIPE_ID
+            );
+        }
+        Recipe.instance = undefined;
+    }
+
+    static getInstanceOrThrowError(): Recipe {
+        if (Recipe.instance !== undefined) {
+            return Recipe.instance;
+        }
+        throw new STError(
+            {
+                type: STError.GENERAL_ERROR,
+                payload: new Error("Initialisation not done. Did you forget to call the SuperTokens.init function?"),
+            },
+            Recipe.RECIPE_ID
+        );
+    }
+
+    getAPIsHandled = (): APIHandled[] => {
+        let apisHandled = [
+            ...this.emailPasswordRecipe.getAPIsHandled(),
+            ...this.emailVerificationRecipe.getAPIsHandled(),
+        ];
+        if (this.thirdPartyRecipe !== undefined) {
+            apisHandled.push(...this.thirdPartyRecipe.getAPIsHandled());
+        }
+        return apisHandled;
+    };
+
+    handleAPIRequest = async (id: string, req: express.Request, res: express.Response, next: express.NextFunction) => {
+        if (id === SIGN_UP_API) {
```

Comment:
Since you are not modifying any API, then you can simply call the sub recipe's `this.emailVerificationRecipe.handleAPIRequest`..

Reply:
done